host-bin/enter-chroot

```diff
@@ -684,10 +684,18 @@ if [ -n "$NOLOGIN" ]; then
         fi
     fi
 else
-    # Run rc.local
+    # Check and run rc.local
     if [ -n "$firstrun" -a -x "$CHROOT/etc/rc.local" ]; then
+        ( 
+            sleep 10
+            echo "Still running chroot's rc.local..." 2>&1
+        ) & forkpid=$! 
         chrootcmd 'exec /etc/rc.local >/dev/null 2>/dev/null </dev/null' \
             || ret=$?
+        addtrap "kill '$forkpid' 2>/dev/null"
+        kill "$forkpid"
+        wait "$forkpid"
+        undotrap
```

Comment:
@dnschneid Updated. Am I using the addtrap and undotrap correctly? should be this way